firedrake/constant.py

```diff
@@ -87,6 +87,10 @@ def evaluate(self, x, mapping, component, index_values):
             return self.dat.data_ro
         return self.dat.data_ro[component]
 
+    def values(self):
+        """Return a (flat) view of the value of the Constant."""
+        return self.dat.data_ro.reshape(-1)
+
     def function_space(self):
         """Return a null function space."""
         return None
```

Comment:
Arguably, we should go back and make `Constant(...) => Function(FunctionSpace(mesh, "Real", 0))`.  But the implementation of the latter does not support vector and tensor-valued things, whereas the former does.